phac-nml/irida-next
# Email notifications for pipelines

## What does this PR do and why?
Implemented emails for notifying the submitter when a pipeline completes or errors.
Email template style copied from #464 for consistency. 
Addresses half of #476.

## Screenshots or screen recordings
![image](https://github.com/phac-nml/irida-next/assets/325703/5053529b-696c-4591-9879-b0b110ce7332)

![image](https://github.com/phac-nml/irida-next/assets/325703/fa8968f7-b5dc-4a7c-96bc-ca81c7acc814)

## How to set up and validate locally
1. Install local mock email service, [Mailhog](https://github.com/mailhog/MailHog).
1. After setup, open a terminal and run `~/go/bin/MailHog`.
1. Run irida-next.
1. Open a rails console.
1. Select a workflow execution.
`w = WorkflowExecution.first`
1. Make sure email notifications are enabled.
`w.email_notification = true`
1. Change state to `completed` or `error`.
`w.state = "completed"`
1. Save the workflow execution.
`w.save`
1. Check mock email service for sent email. Open a browser and navigate to http://0.0.0.0:8025/.
1. Repeat steps for state `error`.


Note: Email previews are located at http://localhost:3000/rails/mailers/pipeline_mailer.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
